src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt

```diff
@@ -164,6 +253,49 @@ class LicenceConditionService(
         }
       }
     }
+
+    /*
+    Using this function, an additional condition is either a new condition or to be removed. Updates to existing
+    additional conditions only involve updating the data associated with the condition using the updateAdditionalConditionData
+    function
+     */
+    val changes: Map<String, Any>
+    if (newAdditionalConditions.isNotEmpty()) {
+      changes = mapOf(
+        "typeOfChange" to "add",
+        "condition" to "additional",
+        "changes" to
+          newAdditionalConditions.map {
+            mapOf(
+              "conditionCode" to it.conditionCode,
+              "conditionType" to it.conditionType,
+            )
+          }
+      )
+    } else {
```

Comment:
I think it's possible for there to be new conditions added and old conditions removed in one POST (ie both `newAdditionalConditions` and `removedAdditionalConditionsList` will have data that we want to audit).
Unless I'm mistaken, when on the screen containing all of the additional conditions and checkboxes if a user checks a new box and unchecks an existing box, it will send both changes in one request.

Reply:
Addressed